### PR TITLE
RA-289: Allow login using only keyboard (without requiring mouse)

### PR DESCRIPTION
**Pull request by collinsnji**
Adding support for keyboard only login to the login page.

### CODE REVIEW

**Comment by dkayiwa** [comment]
@collinsnji how about adding this functionality without changing the styling?

**Comment by collinsnji** [comment]
@dkayiwa I changed the styling so that it is exactly the same as the previous. I used buttons instead of 
the previous list elements and since buttons already had some style attributes I had to add a few more lines to make them identical to the list items.

**Comment by dkayiwa** [comment]
@collinsnji this ticket is simply about keyboard navigation. If you would like to make changes to the style, you will need a separate ticket and pull request because that will also involve discussions about whether the proposed style is in line with the reference application way forward. :)

**Comment by collinsnji** [comment]
OK, @dkayiwa I get the point :slightly_smiling_face: and I'll try making it without changing any style.

**Comment by collinsnji** [comment]
@dkayiwa Done. I updated the work I did and removed the additional styling

File: omod/src/main/webapp/pages/login.gsp [hunk 1]
```diff
@@ -36,18 +36,76 @@ ${ ui.includeFragment("referenceapplication", "infoAndErrorMessages") }
...
+		
+		jQuery('#username').focus();
+		/* 
+		 * Listen for <ENTER> key press and submit form if pressed
+		 */
+		jQuery('#sessionLocation').keyup(function (e) {
+    		var key = e.which || e.keyCode;
+    		if (key === 13 || updateSelectedOption()) {
+      			jQuery('#login-form').submit();
+				console.log("Submitted");
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/login.gsp]: Do you still need the above log?

File: omod/src/main/webapp/pages/login.gsp [hunk 2]
```diff
@@ -36,18 +36,76 @@ ${ ui.includeFragment("referenceapplication", "infoAndErrorMessages") }
...
+					/*
+					 * For the Arrow Up and Arrow Down key press events, 
+					 * Get the value of the <value> attribute of the currently focused <li> and 
+					 * focus a new locations based on that value
+					*/
+					// Up arrow key pressed
+					case 38:
+						switch (jQuery(this).attr("value")) {
+							case 7: jQuery('#sessionLocation li[id="Inpatient Ward"').focus(); break;
+							case 5: jQuery('#sessionLocation li[id="Laboratory"').focus(); break;
```
Comment (dkayiwa) [inline omod/src/main/webapp/pages/login.gsp]: What happens when one has locations different from these ones that you have hard coded?

Comment (dkayiwa) [inline omod/src/main/webapp/pages/login.gsp]: Here is an example of such locations: http://int02.openmrs.org/openmrs/login.htm

Comment (collinsnji) [inline omod/src/main/webapp/pages/login.gsp]: @dkayiwa That's right, I actually taught the locations were constant, but I'll try to see how to address this :slightly_smiling_face:

File: omod/src/main/webapp/pages/login.gsp (hunk 1 above)
Comment (collinsnji) [inline omod/src/main/webapp/pages/login.gsp]: @dkayiwa I'll remove this right away :smiley: I forgot I had that lol

**Comment by collinsnji** [comment]
@dkayiwa After brain storming the whole night, I came up with a solution for N locations. And the build is successful :smile: :fireworks:

**Comment by dkayiwa** [comment]
@collinsnji are you sure you tested this out in a running web application?

**Comment by collinsnji** [comment]
@dkayiwa, I'm so sorry about the error. I forgot to actually replace the `$` sign with `jQuery` I'm fixing it right away

**Comment by dkayiwa** [comment]
@collinsnji this time ensure that you test before committing :)

**Comment by collinsnji** [comment]
@dkayiwa OK. I will. And sorry again about the setback :grimacing:

**Comment by collinsnji** [comment]
@dkayiwa Other than that the code is OK right? :grimacing: I made it support any number of locations
instead of the constant I did before :)

**Comment by dkayiwa** [comment]
Did you intentionally remove focus from the username field on load?

**Comment by collinsnji** [comment]
@dkayiwa lol, I totally forgot, during development, I used the `autofocus` attribute but forgot to replace 
it. I have fixed that though. And also, I made a quick demo [here](https://collinsnji.github.io/refapp-login/)

**Pull request merged**